From b5717c5efa92f8c484aae1839d6eaa7c6d6105f4 Mon Sep 17 00:00:00 2001
From: Ivan Mironov <mironov.ivan@gmail.com>
Date: Thu, 29 Jul 2021 10:29:17 +0500
Subject: [PATCH] Input: synaptics - enable InterTouch for ThinkPad T590

This fixes 3-finger gestures in GNOME.

Tested on

	product: 20N4S1T200 (LENOVO_MT_20N4_BU_Think_FM_ThinkPad T590)
	vendor: LENOVO
	version: ThinkPad T590

Signed-off-by: Ivan Mironov <mironov.ivan@gmail.com>
Cc: stable@vger.kernel.org
---
 drivers/input/mouse/synaptics.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/input/mouse/synaptics.c b/drivers/input/mouse/synaptics.c
index 380aa1614442..1e45bc96b3c7 100644
--- a/drivers/input/mouse/synaptics.c
+++ b/drivers/input/mouse/synaptics.c
@@ -188,6 +188,7 @@ static const char * const smbus_pnp_ids[] = {
 	"LEN2044", /* L470  */
 	"LEN2054", /* E480 */
 	"LEN2055", /* E580 */
+	"LEN205b", /* Lenovo ThinkPad T590 */
 	"LEN2068", /* T14 Gen 1 */
 	"SYN3015", /* HP EliteBook 840 G2 */
 	"SYN3052", /* HP EliteBook 840 G4 */
-- 
2.46.0

